### PR TITLE
bug-46 fix failure to include two bundles with same artifact id and version but different group id

### PR DESCRIPTION
**Pull request by lukejackson**
Test case and fix for bug 46. If two artifacts share the same artifact id and version, they both get written to the same temporary output file, causing one to overwrite the other. the fix is to create all temporary bundles in a subdirectory of the output directory named after the group id.

### CODE REVIEW

**Comment by lukejackson** [comment]
once https://github.com/reficio/p2-maven-plugin/pull/71 is merged i'll rebase this change so the build succeeds

File: src/main/java/org/reficio/p2/P2Helper.java [hunk 1]
```diff
@@ -48,15 +50,25 @@
 public class P2Helper {
 
     public static ArtifactBundlerRequest createBundlerRequest(P2Artifact p2Artifact, ResolvedArtifact resolvedArtifact, File outputFolder) {
-        File binaryInputFile = resolvedArtifact.getArtifact().getFile();
-        File binaryOutputFile = new File(outputFolder, resolvedArtifact.getArtifact().getFile().getName());
+        Artifact artifact = resolvedArtifact.getArtifact();
+        Artifact sourceArtifact = resolvedArtifact.getSourceArtifact();
+
+        File artifactOutputFolder = new File(outputFolder, artifact.getGroupId());
```
Comment (lukejackson) [inline src/main/java/org/reficio/p2/P2Helper.java]: this line is the fix

File: src/test/integration/bug-46-it/site/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                        </goals>
+
+                        <configuration>
+                            <artifacts>
+                                <artifact>
+                                    <id>org.reficio.group1:bundle:1.0.0</id>
+                                </artifact>
+                                <artifact>
+                                    <id>org.reficio.group2:bundle:1.0.0</id>
+                                </artifact>
```
Comment (lukejackson) [inline src/test/integration/bug-46-it/site/pom.xml]: previously group2 bundle would have overwritten group1 bundle, hence group1 bundle would not appear in the update site

**Comment by lukejackson** [comment]
merged master and tests are passing now

**Comment by tombujok** [comment]
Thanks!

**Pull request merged**